### PR TITLE
feat: ced-799 es-ds Update EsAccordion to have two style variants

### PR DESCRIPTION
**Pull request by lazard9**
### 🔗 Linked issue
https://energysage.atlassian.net/browse/CED-799

### ❓ Type of change
- Creating two style variants for EsAccordion component.

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
- I have used variant as prop in the EsAccordion component to enable insertion of different styles for two variants default and minimal. Options are limited to two choices. Default is set by default.
- There is followup ticket [CED-800](https://energysage.atlassian.net/browse/CED-800) already in [the code review](https://github.com/EnergySage/es-ds/pull/1089).

### 🥼 Testing
- Manual testing.

#### 🧐 Feedback Requested / Focus Areas

-

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.
- [x] I have documented testing approach

[CED-800]: https://energysage.atlassian.net/browse/CED-800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ericdouglaspratt — COMMENTED** [review]
The `variant` prop will probably need to be on EsAccordionList rather than EsAccordion, so it can be set one time rather than on every single accordion. You can then pass it into the `provide()` function in EsAccordionList and receive it in the `inject` object in EsAccordion to pass it from EsAccordionList to every EsAccordion underneath it.

You'll also need to merge in from main and resolve the conflicts to receive Su Lei's heading changes.

And I believe the design shows that the "minimal" variant should have no side padding and the heading text should be larger.

**Comment by lazard9** [comment]
> The `variant` prop will probably need to be on EsAccordionList rather than EsAccordion, so it can be set one time rather than on every single accordion. You can then pass it into the `provide()` function in EsAccordionList and receive it in the `inject` object in EsAccordion to pass it from EsAccordionList to every EsAccordion underneath it.
> 
> You'll also need to merge in from main and resolve the conflicts to receive Su Lei's EsAccordion heading changes.
> 
> And I believe the design shows that the "minimal" variant should have no side padding and the heading text should be larger.

Thanks for the review. Have yesterday also realised that I have to inject in child components from the parent ESAccordionList. Hope that all design issues have been fixed. I have incorporated changes from Su Lei PR CED-800.

Please test the solution for any errors and potential hydration problems! :)

File: es-vue-base/src/lib-components/EsAccordion.vue [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-                class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between px-100 px-sm-200 py-100 text-body font-size-100"
+                class="EsAccordion-heading mb-0 align-items-center d-flex font-weight-bold justify-content-between py-100 rounded-0 text-body text-body font-size-100 font-size-lg-300"
                 :class="{
-                    'bg-gray-200': isVisible,
-                    'bg-white': !isVisible,
+                    'bg-gray-200': isVisible && accordionVariant === 'default',
+                    'bg-white': !isVisible && accordionVariant === 'default',
                     'EsAccordion-heading--visible': isVisible,
+                    'px-100 px-sm-200': accordionVariant === 'default',
+                    'px-0': accordionVariant === 'minimal',
```
Comment (suleiy) [inline es-vue-base/src/lib-components/EsAccordion.vue:19]: to make the heading tag larger in minimal variant like in figma,
`'px-0 font-size-200 font-size-lg-400': accordionVariant === 'minimal',`

**Review by suleiy — APPROVED** [review]
Changes look good to me :)
Nit: making the header larger for minimal variant and quick update to test snapshots

File: es-vue-base/src/lib-components/EsAccordion.vue (hunk 1 above)
Comment (suleiy) [inline es-vue-base/src/lib-components/EsAccordion.vue:19]: to make the heading tag larger in minimal variant like in figma,
`'px-0 font-size-200 font-size-lg-400': accordionVariant === 'minimal',`

**Review by ericdouglaspratt — APPROVED** [review]
Looks good! Made a few finalizing changes.

**Pull request merged**